### DISCUSSION TITLE
Will an ActiveRecord `before_create` callback run with Rodauth?

### DISCUSSION THREAD

**Discussion by Samuelodan** [janko/rodauth-rails#266] [General]
Hi there, I'm considering adding support for UUID V7 in my application with the implementation below:
```rb
class ApplicationRecord < ActiveRecord::Base
  primary_abstract_class

  before_create :generate_uuid_v7

  private

    def generate_uuid_v7
      return if self.class.attribute_types['id'].type != :uuid

      self.id ||= UUID7.generate
    end
end
```

Do you think this callback will get triggered automatically after sign-up with rodauth-rails?

**Comment by janko** [comment]
Rodauth doesn't use Active Record, so it will not run any callbacks or validations on the model. For UUIDs, you can generate one in a Rodauth hook:

```rb
before_create_account do
  account[:id] = UUID7.generate
end
```

**Reply by Samuelodan** [reply]
Okay, thank you so much. I was skeptical about doing it in an ActiveRecord callback anyway, so I'm considering installing the pg_uuidv7 extension for Postgres so it's handled at the db level.
